src/query/ast/src/parser/statement.rs

```diff
@@ -2602,6 +2602,15 @@ pub fn hint(i: Input) -> IResult<Hint> {
     rule!(#hint|#invalid_hint)(i)
 }
 
+pub fn top_n(i: Input) -> IResult<u64> {
+    map(
+        rule! {
+            TOP ~ ^#literal_u64
+        },
+        |(_, n)| n,
+    )(i)
+}
```

Comment:
```suggestion
pub fn top_n(i: Input) -> IResult<u64> {
    map_res(
        rule! {
            TOP ~ #literal_u64? : "TOP <n>"
        },
        |(_, opt_n)| {
            if let Some(n) = opt_n {
                Ok(n)
            } else {
                Err(nom::Err::Failure(ErrorKind::Other(
                    "expecting a literal number after keyword `TOP`, if you were refering a column with name `top`, \
                    please quote it like `\"top\"`"
                )))
            }
        },
    )(i)
}
```

Reply:
Also add `select top from t1` to `test_statement_error()` in `src/query/ast/tests/it/parser.rs`

Reply:
Before merging I missing this comemnts, cc @TCeason

Reply:
> Before merging, I missed these comments. cc @TCeason

It's ok, Andy and I push for this in commit:

https://github.com/datafuselabs/databend/pull/15318/commits/1c7cda284c43ab64d431bc96acc2c4d44b27a324
https://github.com/datafuselabs/databend/pull/15318/commits/db3cf37dfceb334cbf2fa27dceebf291d9ea4d30